cnxepub/scripts/collated_single_html/main.py

```diff
@@ -5,31 +5,51 @@
 # Public License version 3 (AGPLv3).
 # See LICENCE.txt for details.
 # ###
-"""\
-Processes a collated book's single-html representation for validation
-and parser of specific elements.
+"""Processes a collated book's single-html representation for validation.
 
+Ensures safe parsing for specific elements.
 """
 import argparse
-import os
+import logging
 import sys
 from pprint import pprint
-
+from lxml import etree
 import cnxepub
 
+logger = logging.getLogger('cnxepub')
+
+formatter = logging.Formatter('%(name)s %(levelname)s %(message)s')
+handler = logging.StreamHandler(sys.stderr)
+handler.setFormatter(formatter)
+logger.addHandler(handler)
+
 
 def main(argv=None):
+    """Parse passed in cooked single HTML."""
     parser = argparse.ArgumentParser(description=__doc__)
     parser.add_argument('collated_html', type=argparse.FileType('r'),
-                        help='Path to the collated html file (use - for stdin)')
+                        help='Path to the collated html'
+                             ' file (use - for stdin)')
+    parser.add_argument('-d', '--dump-tree', action='store_true',
+                        help='Print out parsed model tree.')
+
     args = parser.parse_args(argv)
 
-    binder = cnxepub.adapt_single_html(args.collated_html.read())
-    pprint(cnxepub.model_to_tree(binder))
+    html = args.collated_html.read()
+    try:
+        htree = etree.fromstring(html)
+    except etree.XMLSyntaxError:
+        htree = etree.HTML(html)
```

Comment:
IMO there's no point in doing this in the validator script and then failing when we actually reconstitute.